### PR TITLE
Include courier crediting in deposit function

### PR DESCRIPTION
**Pull request by haydenshively**
It was really annoying to have to call a separate function (`creditCourier`) before depositing. This fixes that, although it does add a bit of complexity.

### CODE REVIEW

File: core/test/LenderReferrals.t.sol
Comment (aneesh1) [inline core/test/LenderReferrals.t.sol:1]: Tests look good! Two suggestions:
1. Should we change the name of this file from `LenderReferrals.t.sol` to `LenderRewards.t.sol`? 
2. Do we have a test where we `deposit` into an account with 0 balance along with a courier? Based on my understanding of `_mint`, this would set the courier accordingly.

**Review by aneesh1 — APPROVED** [review]
Streamlines the whole process - nice!

File: core/test/LenderReferrals.t.sol
Comment (haydenshively) [reply core/test/LenderReferrals.t.sol:1]: 1. It's still primarily testing referrals, rewards are separate (though everything does kinda happen inside of `_mint` now)
2. Yes! [Here](https://github.com/aloelabs/aloe-ii/blob/a84d1501ddce45d611fc34966544025a632645fd/core/test/LenderReferrals.t.sol#L100) and also the invariants cover this

File: core/test/gas/LenderGas.t.sol [hunk 1]
```diff
@@ -59,7 +59,7 @@ contract LenderGasTest is Test {
         vm.prank(alice);
         asset.approve(address(this), type(uint256).max);
 
-        // `alice` allows this test contract to manage his WETH+
+        // `alice` allows this test contract to manage her WETH+
```
Comment (IanWoodard) [inline core/test/gas/LenderGas.t.sol:62]: nice catch 😂

File: core/src/RateModel.sol [hunk 2]
```diff
@@ -3,7 +3,7 @@ pragma solidity 0.8.17;
 
 import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
 
-import {MAX_RATE, ONE} from "./libraries/constants/Constants.sol";
+import {MAX_RATE, ONE} from "src/libraries/constants/Constants.sol";
```
Comment (IanWoodard) [inline core/src/RateModel.sol]: Why the change?

**Review by IanWoodard — APPROVED** [review]

File: core/src/RateModel.sol (hunk 2 above)
Comment (haydenshively) [reply core/src/RateModel.sol]: Good catch. Messed up the merge.
